### ISSUE TITLE
`micropkg` cannot package modules that are not directly under the `src/<project_name>/` directory

### ISSUE THREAD

**Issue by lamhm** [kedro-org/kedro#2200]
## Description

`micropkg` cannot package sub-folders that are more than 1 level below the `src/<project_name>/` folder. This has rendered `micropkg` unusable for pipelines (if you follow the standard folder structure of Kedro).

## Steps to Reproduce

Here was my `src` folder structure:

```plaintext
src
|-- <project_name>
|   |-- io
|   |   |-- __init__.py
|   |   |-- ...
|   |-- pipelines
|       |-- ...
|       |-- clean_data
|           |-- __init__.py
|           |-- ...
|-- tests
|   |-- ...
|-- ...
```

Under the `[tool.kedro.micropkg.package]` spec of `pyproject.toml`, I had:

```plaintext
[tool.kedro.micropkg.package]
io = {}
pipelines.clean_data = {}
```

Then, I executed `kedro micropkg package --all`.

## Expected Result

Both the `io` and the `pipelines.clean_data` modules should be packaged without problems.

## Actual Result

Only the `io` module was packaged successfully.

Packaging the `pipelines.clean_data` module raised the following error:

```plaintext
TypeError: _package_micropkg() got an unexpected keyword argument 'clean_data'
```

## My Environment

* Kedro version used: 0.18.4
* Python version used: 3.10.8
* Operating system and version: Ubuntu 20.04

## Important Note

To make `micropkg` works on Kedro 0.18.4, I have manually fixed the bug #2119 on my local environment (by following what @merelcht has described in the [bug report](https://github.com/kedro-org/kedro/pull/2119#issue-1494312175)). Without this step, `micropkg` would not work at all.

**Comment by noklam** [comment]
@lamhm Is this still failing or fixed by #2119 already?

**Comment by noklam** [comment]
Closing this as I believe this is already fixed in previous releases, feel free to re-open if it isn't the case.

**Comment by lamhm** [comment]
@noklam: Apologies for my late response.

While testing Kedro 0.18.6, I have just noticed that I forgot to quote the key `pipelines.clean_data` in the `[tool.kedro.micropkg.package]` section of `project.toml`. As soon as I surrounded `pipelines.clean_data` with quotation marks, `micropkg` worked perfectly.

So, [this](https://github.com/kedro-org/kedro/issues/2200#issue-1530298964) may be an invalid bug report. Sorry for having bothered you. And thank you very much for your support.
